### PR TITLE
Alcoholic Quirk

### PR DESCRIPTION
**Pull request by That0nePerson**
## About The Pull Request
Adds an alcoholic quirk that gives the holder mood debuffs (scales over time between drinks) if they're not drunk. -1 cost

## Why It's Good For The Game
Dino_Horse on discord wanted it, allows you to have a good reason to drink and hang around the bar or RP an alcoholic character

## Changelog
:cl:
add: Added alcoholic qurik
/:cl:

### CODE REVIEW

**Comment by eeSPee** [comment]
>if(prob(30))
>to_chat(H, "<span class='boldannounce'>You're not feeling good at all! You really need some alcohol.</span>")

Prob 30 may be a little too much and you may end up being spammed by these messages.

**Comment by That0nePerson** [comment]
I turned the prob down to 5, thanks for the tip

File: code/datums/traits/negative.dm [hunk 1]
```diff
@@ -559,6 +559,66 @@
...
+	drink_instance = new drink_instance()
+	var/list/slots = list("in your backpack" = SLOT_IN_BACKPACK)
+	var/mob/living/carbon/human/H = quirk_holder
+	where_drink = H.equip_in_one_of_slots(drink_instance, slots, FALSE) || "at your feet"
+
+/datum/quirk/alcoholic/post_add()
+	to_chat(quirk_holder, "<span class='boldnotice'>There is a small bottle of [drink_instance] [where_drink]. You only have a single bottle, might have to find some more...</span>")
+
+datum/quirk/alcoholic/on_process()
+	if(tick_number >= check_tick)
```
Comment (PowerfulBacon) [inline code/datums/traits/negative.dm]: Is check_tick ever changed?

Comment (That0nePerson) [reply code/datums/traits/negative.dm]: nah idk why I made it

File: code/datums/traits/negative.dm [hunk 2]
```diff
@@ -559,6 +559,65 @@
...
+			if(10 to 20)
+				SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "alcoholic", /datum/mood_event/withdrawal_medium, "alcohol")
+				if(prob(5))
+					to_chat(H, "<span class='notice'>You feel like you need alcohol. You just can't stand being sober.</span>")
+			if(20 to 30)
+				SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "alcoholic", /datum/mood_event/withdrawal_severe, "alcohol")
+				if(prob(5))
+					to_chat(H, "<span class='danger'>You have an intense craving for a drink.</span>")
+			if(30 to INFINITY)
+				SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "alcoholic", /datum/mood_event/withdrawal_critical, "Alcohol")
```
Comment (PowerfulBacon) [inline code/datums/traits/negative.dm:614]: The alcohol is capitalized on this one, but not all the others. Not sure if that affects anything in game. Also the "alcoholic" mood event source should be a define for consistency

Comment (That0nePerson) [reply code/datums/traits/negative.dm:614]: the alcohol is capitalized on this one cuz the message is "[input]! [input]! [input]!" so it doesn't look right if it's not capitalized

Comment (That0nePerson) [reply code/datums/traits/negative.dm:614]: and we talked about this on discord, no other mood even sources I can see have defines

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**